modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/ExecutionServiceImpl.java

```diff
@@ -536,27 +554,21 @@ private QueryPlan prepareSingle(SqlNode sqlNode, PlanningContext ctx) throws Val
 
         ctx.planner().reset();
 
-        switch (sqlNode.getKind()) {
-            case SELECT:
-            case ORDER_BY:
-            case WITH:
-            case VALUES:
-            case UNION:
-                return prepareQuery(sqlNode, ctx);
+       if (sqlNode.getKind().belongsTo(SqlKind.QUERY))
```

Comment:
this Set is not equal with previous, is it ok here?

Reply:
hm, it's better to return the old logic